include/revng/Model/TupleTree.h

```diff
@@ -448,6 +457,15 @@ struct CallByPathVisitorWithInstance {
   }
 
   template<typename T, typename K, typename KeyT>
+  requires IsUpcastablePointer<K>
```

Comment:
```suggestion
  template<typename T, IsUpcastablePointer K, typename KeyT>
```

Reply:
Done. I also added another `concept NotIsUpcastablePointer` to use for consistency below, where I previously had `requires (not IsUpcastablePointer<K>)`.

Reply:
`s/NotIsUpcastablePointer/IsNotUpcastablePointer/` maybe

Reply:
Ok. Done.